spring-boot-project/spring-boot-actuator/src/main/java/org/springframework/boot/actuate/cassandra/CassandraDriverHealthIndicator.java

```diff
@@ -52,11 +52,18 @@ public CassandraDriverHealthIndicator(CqlSession session) {
 
 	@Override
 	protected void doHealthCheck(Health.Builder builder) throws Exception {
-		Row row = this.session.execute(SELECT).one();
-		builder.up();
-		if (row != null && !row.isNull(0)) {
-			builder.withDetail("version", row.getString(0));
+		Collection<Node> nodes = this.session.getMetadata().getNodes().values();
+		boolean atLeastOneUp = nodes.stream().map(Node::getState).anyMatch((state) -> state == NodeState.UP);
+		if (atLeastOneUp) {
```

Comment:
This if/else branch should be simplified with a ternary operator using `builder.status`.